blocks/api/raw-handling/test/special-comment-converter.js

```diff
@@ -12,38 +24,75 @@ import { deepFilterHTML } from '../utils';
 describe( 'specialCommentConverter', () => {
 	it( 'should convert a single comment into a basic block', () => {
 		equal(
-			deepFilterHTML( '<!--more-->', [ specialCommentConverter ] ),
-			'<wp-block data-block="core/more"></wp-block>'
+			deepFilterHTML( wrapInP(
+				'<!--more-->'
+			), [ specialCommentConverter ] ),
+			withEmptyP( '<wp-block data-block="core/more"></wp-block>' )
 		);
 	} );
 	it( 'should convert two comments into a block', () => {
 		equal(
-			deepFilterHTML( '<!--more--><!--noteaser-->', [ specialCommentConverter ] ),
-			'<wp-block data-block="core/more" data-no-teaser=""></wp-block>'
+			deepFilterHTML( wrapInP(
+				'<!--more--><!--noteaser-->'
+			), [ specialCommentConverter ] ),
+			withEmptyP( '<wp-block data-block="core/more" data-no-teaser=""></wp-block>' )
 		);
 	} );
 	it( 'should pass custom text to the block', () => {
 		equal(
-			deepFilterHTML(
-				'<!--more Read all about it!--><!--noteaser-->',
-				[ specialCommentConverter ]
+			deepFilterHTML( wrapInP(
+				'<!--more Read all about it!--><!--noteaser-->'
+			), [ specialCommentConverter ]
 			),
-			'<wp-block data-block="core/more" data-custom-text="Read all about it!" data-no-teaser=""></wp-block>'
+			withEmptyP( '<wp-block data-block="core/more" data-custom-text="Read all about it!" data-no-teaser=""></wp-block>' )
 		);
 	} );
-	it( 'should handle reformatted content', () => {
+	it( 'should not break content order', () => {
 		const output = deepFilterHTML(
-			`<p>
-				<!--more-->
-				<!--noteaser-->
-			</p>`,
+			`<p>First paragraph.<!--more--></p>
+			<p>Second paragraph</p>
+			<p>Third paragraph</p>`,
 			[ specialCommentConverter ]
 		);
-		// Skip the empty paragraph, which other transforms would eliminate
-		const start = output.indexOf( '</p>' ) + '</p>'.length;
 		equal(
-			output.substr( start ),
-			'<wp-block data-block="core/more" data-no-teaser=""></wp-block>'
+			trimWhitespace( output ),
+			trimWhitespace( `<p>First paragraph.</p>
```

Comment:
I don't understand why `trimWhitespace` is necessary. Shouldn't it be preserved?

Reply:
(I mean, isn't `specialCommentConverter` merely moving other nodes?)

Reply:
Yeah. As I was developing I wanted the tests to have these `input`/`output` strings neatly aligned with the rest of the code but that meant adding tab characters throughout. Not that that's wrong, but it made the tests more brittle. I'll probably remove this.